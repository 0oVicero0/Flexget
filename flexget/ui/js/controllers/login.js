'use strict';

app.requires.push('http-auth-interceptor');

app.controller('LoginController', function ($scope, $http, $mdDialog, authService) {
  $scope.error = '';

  var doLogin = function() {
    var LoginModalController = function($scope) {
      $scope.error = '';
      $scope.credentials = {
        username: '',
        password: ''
      };
      $scope.login = function () {
        $http.post('/api/login/', $scope.credentials).success(function() {
          authService.loginConfirmed();
        }).error(function(data) {
          $scope.credentials.password = '';
          if ('message' in data) {
            $scope.error = data.message;
          } else {
            $scope.error = 'Error during authentication';
          }
        });
      }
    };
    $mdDialog.show({
      templateUrl: 'static/partials/login.html',
      parent: angular.element(document.body),
      controller: LoginModalController
    });

  };

  var closeLogin = function() {
    $mdDialog.hide();
  };

  $scope.$on('event:auth-loginRequired', doLogin);
  $scope.$on('event:auth-loginConfirmed', closeLogin)
});
