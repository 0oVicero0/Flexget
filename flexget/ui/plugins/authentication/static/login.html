<div class="modal-header">
    <h3>Login</h3>
</div>
<div class="modal-body">
    <p class="text-danger">{{ error }}</p>
    <form>
        <div class="form-group">
            <label for="username">Username</label>
            <input type="text" class="form-control" id="username" placeholder="Username" ng-model="credentials.username">
        </div>
        <div class="form-group">
            <label for="password">Password</label>
            <input type="password" class="form-control" id="password" placeholder="Password" ng-model="credentials.password">
        </div>
    </form>
</div>
<div class="modal-footer">
    <button type="button" class="btn btn-success" data-ng-click="login()">Login</button>
</div>