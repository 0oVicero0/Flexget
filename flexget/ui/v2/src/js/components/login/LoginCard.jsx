import React from 'react';
import PropTypes from 'prop-types';
import { Field, reduxForm } from 'redux-form';
import Button from 'material-ui/Button';
import Card, { CardActions, CardContent } from 'material-ui/Card';
import { withStyles, createStyleSheet } from 'material-ui/styles';
import TextField from 'components/common/TextField';

const styleSheet = createStyleSheet('LoginCard', theme => ({
  card: {
    maxWidth: 400,
    margin: '0 10px',
    [theme.breakpoints.up('sm')]: {
      margin: '0 auto',
    },
  },
  error: {
    color: theme.palette.error[500],
    textAlign: 'center',
    padding: 10,
  },
  button: {
    width: '100%',
  },
}));

const LoginCard = ({ classes, handleSubmit, errorStatus }) => (
  <Card className={classes.card}>
    <form onSubmit={handleSubmit}>
      <CardContent>
        <div className={classes.error}>
          {errorStatus.message}
        </div>
        <Field
          name="username"
          component={TextField}
          id="username"
          label="Username"
        />
        <Field
          name="password"
          component={TextField}
          id="password"
          label="Password"
          type="Password"
        />
      </CardContent>
      <CardActions>
        <Button type="submit" className={classes.button}>
          Login
        </Button>
      </CardActions>
    </form>
  </Card>
);

LoginCard.propTypes = {
  classes: PropTypes.object.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  errorStatus: PropTypes.object,
};

LoginCard.defaultProps = {
  errorStatus: {},
};

export default reduxForm({
  form: 'login',
})(withStyles(styleSheet)(LoginCard));
