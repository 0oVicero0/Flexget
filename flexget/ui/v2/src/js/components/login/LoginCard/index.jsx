import React from 'react';
import PropTypes from 'prop-types';
import { Field, reduxForm } from 'redux-form';
import Button from 'material-ui/Button';
import Card, { CardActions, CardContent } from 'material-ui/Card';
import { withStyles } from 'material-ui/styles';
import TextField from 'components/common/TextField';

const styleSheet = theme => ({
  card: {
    maxWidth: 400,
    margin: '0 10px',
    [theme.breakpoints.up('sm')]: {
      margin: '0 auto',
    },
  },
  error: {
    color: theme.palette.error[500],
    textAlign: 'center',
    padding: 10,
  },
  button: {
    width: '100%',
  },
  cardContent: {
    display: 'flex',
    flex: {
      direction: 'column',
    },
  },
  field: {
    paddingBottom: 10,
  },
});

export const LoginCard = ({ classes, handleSubmit, errorStatus }) => (
  <Card className={classes.card}>
    <form onSubmit={handleSubmit}>
      <CardContent className={classes.cardContent}>
        <div className={classes.error}>
          {errorStatus.message}
        </div>
        <Field
          name="username"
          component={TextField}
          id="username"
          label="Username"
          className={classes.field}
        />
        <Field
          name="password"
          component={TextField}
          id="password"
          label="Password"
          type="Password"
          className={classes.field}
        />
      </CardContent>
      <CardActions>
        <Button type="submit" className={classes.button}>
          Login
        </Button>
      </CardActions>
    </form>
  </Card>
);

LoginCard.propTypes = {
  classes: PropTypes.object.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  errorStatus: PropTypes.object,
};

LoginCard.defaultProps = {
  errorStatus: {},
};

export default reduxForm({
  form: 'login',
})(withStyles(styleSheet)(LoginCard));
