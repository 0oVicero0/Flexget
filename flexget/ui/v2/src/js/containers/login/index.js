import { connect } from 'react-redux';
import LoginPage from 'components/login';

export function mapStateToProps({ auth }) {
  return {
    redirectToReferrer: !!auth.loggedIn,
  };
}

export default connect(mapStateToProps)(LoginPage);
