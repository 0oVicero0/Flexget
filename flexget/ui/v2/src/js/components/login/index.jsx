import React from 'react';
import PropTypes from 'prop-types';
import { Redirect } from 'react-router-dom';
import headerImage from 'images/header.png';
import { withStyles, createStyleSheet } from 'material-ui/styles';
import LoginCard from 'containers/login/LoginCard';

const styleSheet = createStyleSheet('LoginPage', theme => ({
  logo: {
    background: `transparent url(${headerImage}) no-repeat center`,
    minHeight: 90,
    backgroundSize: '100% auto',
    margin: '0 10px',
    [theme.breakpoints.up('sm')]: {
      backgroundSize: 'auto',
    },
  },
}));

const LoginPage = ({ classes, redirectToReferrer, location }) => {
  const { from } = location.state || { from: { pathname: '/' } };

  if (redirectToReferrer) {
    return (
      <Redirect to={from} />
    );
  }

  return (
    <div>
      <div className={classes.logo} />
      <LoginCard />
    </div>
  );
};

LoginPage.propTypes = {
  classes: PropTypes.object.isRequired,
  location: PropTypes.object.isRequired,
  redirectToReferrer: PropTypes.bool.isRequired,
};

export default withStyles(styleSheet)(LoginPage);
