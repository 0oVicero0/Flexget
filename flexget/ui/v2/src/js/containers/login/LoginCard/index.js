import { connect } from 'react-redux';
import { login } from 'actions/auth';
import LoginCard from 'components/login/LoginCard';

export function mapStateToProps({ status }) {
  return {
    initialValues: {
      username: 'flexget',
    },
    errorStatus: status.error || {},
  };
}

function mapDispatchToProps(dispatch) {
  return {
    onSubmit: credentials => dispatch(login(credentials)),
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(LoginCard);
