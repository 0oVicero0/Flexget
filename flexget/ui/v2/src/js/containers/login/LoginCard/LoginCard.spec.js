import { mapStateToProps } from 'containers/login/LoginCard';

describe('containers/login/LoginCard', () => {
  it('should return no errors if there are none', () => {
    expect(mapStateToProps({ status: { } })).toMatchSnapshot();
  });

  it('should return errors if there are errors', () => {
    expect(mapStateToProps({ status: { error: { message: 'Invalid Credentials' } } })).toMatchSnapshot();
  });
});
