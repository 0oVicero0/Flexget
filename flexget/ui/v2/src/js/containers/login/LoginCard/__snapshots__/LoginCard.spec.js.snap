// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/login/LoginCard should return errors if there are errors 1`] = `
Object {
  "errorStatus": Object {
    "message": "Invalid Credentials",
  },
  "initialValues": Object {
    "username": "flexget",
  },
}
`;

exports[`containers/login/LoginCard should return no errors if there are none 1`] = `
Object {
  "errorStatus": Object {},
  "initialValues": Object {
    "username": "flexget",
  },
}
`;
